backend/src/federation/client/Client.ts

```diff
@@ -0,0 +1,57 @@
+import { FederatedCommunity as DbFederatedCommunity } from '@entity/FederatedCommunity'
+
+import { ApiVersionType } from '@/federation/enum/apiVersionType'
+
+// eslint-disable-next-line camelcase
+import { Client_1_0 } from './Client_1_0'
+// eslint-disable-next-line camelcase
+import { Client_1_1 } from './Client_1_1'
+
+// eslint-disable-next-line camelcase
+type FederationClient = Client_1_0 | Client_1_1
+
+type ClientInstance = {
+  id: number
+  // eslint-disable-next-line no-use-before-define
+  client: FederationClient
+}
+
+export class Client {
+  private static instanceArray: ClientInstance[] = []
+
+  /**
+   * The Singleton's constructor should always be private to prevent direct
+   * construction calls with the `new` operator.
+   */
+  // eslint-disable-next-line no-useless-constructor, @typescript-eslint/no-empty-function
+  private constructor() {}
+
+  private static createFederationClient = (dbCom: DbFederatedCommunity) => {
+    switch (dbCom.apiVersion) {
+      case ApiVersionType.V1_0:
+        return new Client_1_0(dbCom)
+      case ApiVersionType.V1_1:
+        return new Client_1_1(dbCom)
+      default:
+        return null
+    }
+  }
+
+  /**
+   * The static method that controls the access to the singleton instance.
+   *
+   * This implementation let you subclass the Singleton class while keeping
+   * just one instance of each subclass around.
+   */
+  public static getInstance(dbCom: DbFederatedCommunity): FederationClient | null {
+    const instance = Client.instanceArray.find((instance) => instance.id === dbCom.id)
```

Comment:
yes you can decide to avoid pretended calculations. 
But as i mentioned with the invocation of startDHT the home-community entries will be recreated and will get new ids. An update could be insufficient, in case of changed amount of api-versions the amount for home-communty entries will vary and would cause a more difficult calculation, which you want to avoid.
If the backend will not be restarted at the same time, you will get additional instances in the singleton federation-client list for these new home-community entries, because of their new ids even though they act for the same home-community api version.